# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: accessible/ipc/RemoteAccessibleBase.h
# Commit: 95a2cae69fb9
# Full Hash: 95a2cae69fb9d3763a8a8aba8beba2a78fdd570b
# Author: James Teh <jteh@mozilla.com>
# Date: 2022-05-05 16:11:56
# Regressor Bug: 1766794
# File Overlap Count: 2
# Description:
#   Bug 1766794: Lazily cache HyperText offsets for RemoteAccessibles. r=eeejay
#   
#   We already had a similar cache in local HyperTextAccessible.
#   This improves performance significantly when walking the text attributes of a container with a large number of text leaf children, such as is encountered when using view source.
#   This patch unifies that cache across local and remote.
# ==============================================================================

diff -r 3f8a611b454e -r 95a2cae69fb9 accessible/ipc/RemoteAccessibleBase.h
--- a/accessible/ipc/RemoteAccessibleBase.h	Thu May 05 10:56:32 2022 +0000
+++ b/accessible/ipc/RemoteAccessibleBase.h	Thu May 05 11:14:12 2022 +0000
@@ -291,6 +291,12 @@
   // HyperTextAccessibleBase
   virtual already_AddRefed<AccAttributes> DefaultTextAttributes() override;
 
+  virtual void InvalidateCachedHyperTextOffsets() override {
+    if (mCachedFields) {
+      mCachedFields->Remove(nsGkAtoms::offset);
+    }
+  }
+
  protected:
   RemoteAccessibleBase(uint64_t aID, Derived* aParent,
                        DocAccessibleParent* aDoc, role aRole, AccType aType,
@@ -331,6 +337,8 @@
 
   nsAtom* GetPrimaryAction() const;
 
+  virtual const nsTArray<int32_t>& GetCachedHyperTextOffsets() const override;
+
  private:
   uintptr_t mParent;
   static const uintptr_t kNoParent = UINTPTR_MAX;
